lib/framework/notification_manager.dart

```diff
@@ -120,17 +125,15 @@ class NotificationManager {
 
   Future<void> _handleNotification() async {
     Map<String, dynamic>? messageData = Ensemble.externalDataContext['data'];
-    if (messageData?['screenId'] != null ||
-        messageData?['screenName'] != null) {
-      ScreenController().navigateToScreen(
-        Utils.globalAppKey.currentContext!,
+    // If there is a screen, it navigates to that page
+    // or else it'll open the app with the root screen and clears all the previous screens
+    final screenNotFound =
+        messageData?['screenId'] == null && messageData?['screenName'] == null;
+    ScreenController().navigateToScreen(Utils.globalAppKey.currentContext!,
         screenId: messageData!['screenId'],
         screenName: messageData['screenName'],
         pageArgs: messageData,
-      );
-    } else {
-      log('No screenId nor screenName provided on the notification. Ignoring ...');
-    }
+        routeOption: screenNotFound ? RouteOption.clearAllScreens : null);
```

Comment:
i know we talked about this but it's probably not safe to clear all the screens. It's OK to have duplicate screens